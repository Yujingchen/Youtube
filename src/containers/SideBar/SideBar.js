import React, { Component } from "react";
import SideBarItem from "./SideBarItem/SideBarItem";
import { Menu, Divider } from "semantic-ui-react";
import "./SideBar.scss";
import SideBarHeader from "./SideBarHeader/SideBarHeader";
import Subscriptions from "./Subscriptions/Subscriptions";
import SideBarFooter from "./SideBarFooter/SideBarFooter";
export default class SideBar extends Component {
  render() {
    return (
      //Use samantic Menu to build vertical and fixed left sidebar
      <Menu borderless vertical stackable fixed="left" className="side-nav">
        <SideBarItem highlight={true} label="Home" icon="home" />
        <SideBarItem highlight={false} label="Trending" icon="fire" />
        <SideBarItem highlight={false} label="Followers" icon="spy" />
        <Divider />
        <SideBarHeader title="Library" />
        <SideBarItem highlight={false} label="History" icon="history" />
        <SideBarItem highlight={false} label="Watch later" icon="clock" />
        <SideBarItem highlight={false} label="Liked videos" icon="thumbs up" />
        <Divider />
        <Subscriptions />
        <SideBarHeader title="More from Youtube" />
        <SideBarItem highlight={false} label="Movies and Shows" icon="film" />
        <Divider />
        <SideBarItem highlight={false} label="Report history" icon="flag" />
        <SideBarItem highlight={false} label="Help" icon="help circle" />
        <SideBarItem highlight={false} label="Send feedback" icon="comment" />
        <Divider />
        <SideBarFooter />
      </Menu>
    );
  }
}
