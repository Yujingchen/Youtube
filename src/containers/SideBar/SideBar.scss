@import "../../styles/shared.scss";
.ui.menu.fixed.side-nav {
  background-color: #f5f5f5;
  margin-top: $header-nav-height;
  overflow: auto;
  padding-bottom: $header-nav-height;

  .sidebar-item:hover {
    background: #ebebeb;
  }
}

.sidebar-nav.ui.vertical.menu {
  width: 17rem;
}
