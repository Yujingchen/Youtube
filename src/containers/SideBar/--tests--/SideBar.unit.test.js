import React from "react";
import { shallow } from "enzyme";
import SideBar from "../SideBar";
describe("SideBar", () => {
  test("renders SideBar", () => {
    const wrapper = shallow(<SideBar />);
    expect(wrapper).toMatchSnapshot();
  });

  // test("renders SideBarHeader with props.title=''", () => {
  //   const wrapper = shallow(<SideBarHeader title="" />);
  //   expect(wrapper).toMatchSnapshot();
  // });

  // test("renders SideBarHeader with sample title", () => {
  //   const wrapper = shallow(<SideBarHeader title="Sample Title" />);
  //   expect(wrapper).toMatchSnapshot();
  // });
});
