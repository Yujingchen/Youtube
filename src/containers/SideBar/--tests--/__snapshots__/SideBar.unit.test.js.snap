// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders SideBar 1`] = `
<Menu
  borderless={true}
  className="side-nav"
  fixed="left"
  stackable={true}
  vertical={true}
>
  <SideBarItem
    highlight={true}
    icon="home"
    label="Home"
  />
  <SideBarItem
    highlight={false}
    icon="fire"
    label="Trending"
  />
  <SideBarItem
    highlight={false}
    icon="spy"
    label="Followers"
  />
  <Divider />
  <SideBarHeader
    title="Library"
  />
  <SideBarItem
    highlight={false}
    icon="history"
    label="History"
  />
  <SideBarItem
    highlight={false}
    icon="clock"
    label="Watch later"
  />
  <SideBarItem
    highlight={false}
    icon="thumbs up"
    label="Liked videos"
  />
  <Divider />
  <Subscriptions />
  <SideBarHeader
    title="More from Youtube"
  />
  <SideBarItem
    highlight={false}
    icon="film"
    label="Movies and Shows"
  />
  <Divider />
  <SideBarItem
    highlight={false}
    icon="flag"
    label="Report history"
  />
  <SideBarItem
    highlight={false}
    icon="help circle"
    label="Help"
  />
  <SideBarItem
    highlight={false}
    icon="comment"
    label="Send feedback"
  />
  <Divider />
  <SideBarFooter />
</Menu>
`;

exports[`SideBarHeader renders SideBar 1`] = `
<Menu
  borderless={true}
  className="side-nav"
  fixed="left"
  stackable={true}
  vertical={true}
>
  <SideBarItem
    highlight={true}
    icon="home"
    label="Home"
  />
  <SideBarItem
    highlight={false}
    icon="fire"
    label="Trending"
  />
  <SideBarItem
    highlight={false}
    icon="spy"
    label="Followers"
  />
  <Divider />
  <SideBarHeader
    title="Library"
  />
  <SideBarItem
    highlight={false}
    icon="history"
    label="History"
  />
  <SideBarItem
    highlight={false}
    icon="clock"
    label="Watch later"
  />
  <SideBarItem
    highlight={false}
    icon="thumbs up"
    label="Liked videos"
  />
  <Divider />
  <Subscriptions />
  <SideBarHeader
    title="More from Youtube"
  />
  <SideBarItem
    highlight={false}
    icon="film"
    label="Movies and Shows"
  />
  <Divider />
  <SideBarItem
    highlight={false}
    icon="flag"
    label="Report history"
  />
  <SideBarItem
    highlight={false}
    icon="help circle"
    label="Help"
  />
  <SideBarItem
    highlight={false}
    icon="comment"
    label="Send feedback"
  />
  <Divider />
  <SideBarFooter />
</Menu>
`;

exports[`SideBarHeader renders SideBarHeader with props.title='' 1`] = `
<MenuItem>
  <MenuHeader
    className="side-bar-header"
  />
</MenuItem>
`;

exports[`SideBarHeader renders SideBarHeader with sample title 1`] = `
<MenuItem>
  <MenuHeader
    className="side-bar-header"
  >
    SAMPLE TITLE
  </MenuHeader>
</MenuItem>
`;

exports[`renders HeaderNav 1`] = `
<Menu
  borderless={true}
  className="side-nav"
  fixed="left"
  stackable={true}
  vertical={true}
>
  <SideBarItem
    highlight={true}
    icon="home"
    label="Home"
  />
  <SideBarItem
    highlight={false}
    icon="fire"
    label="Trending"
  />
  <SideBarItem
    highlight={false}
    icon="spy"
    label="Followers"
  />
  <Divider />
  <SideBarHeader
    title="Library"
  />
  <SideBarItem
    highlight={false}
    icon="history"
    label="History"
  />
  <SideBarItem
    highlight={false}
    icon="clock"
    label="Watch later"
  />
  <SideBarItem
    highlight={false}
    icon="thumbs up"
    label="Liked videos"
  />
  <Divider />
  <Subscriptions />
  <SideBarHeader
    title="More from Youtube"
  />
  <SideBarItem
    highlight={false}
    icon="film"
    label="Movies and Shows"
  />
  <Divider />
  <SideBarItem
    highlight={false}
    icon="flag"
    label="Report history"
  />
  <SideBarItem
    highlight={false}
    icon="help circle"
    label="Help"
  />
  <SideBarItem
    highlight={false}
    icon="comment"
    label="Send feedback"
  />
  <Divider />
  <SideBarFooter />
</Menu>
`;

exports[`renders SideBar 1`] = `
<Menu
  borderless={true}
  className="side-nav"
  fixed="left"
  stackable={true}
  vertical={true}
>
  <SideBarItem
    highlight={true}
    icon="home"
    label="Home"
  />
  <SideBarItem
    highlight={false}
    icon="fire"
    label="Trending"
  />
  <SideBarItem
    highlight={false}
    icon="spy"
    label="Followers"
  />
  <Divider />
  <SideBarHeader
    title="Library"
  />
  <SideBarItem
    highlight={false}
    icon="history"
    label="History"
  />
  <SideBarItem
    highlight={false}
    icon="clock"
    label="Watch later"
  />
  <SideBarItem
    highlight={false}
    icon="thumbs up"
    label="Liked videos"
  />
  <Divider />
  <Subscriptions />
  <SideBarHeader
    title="More from Youtube"
  />
  <SideBarItem
    highlight={false}
    icon="film"
    label="Movies and Shows"
  />
  <Divider />
  <SideBarItem
    highlight={false}
    icon="flag"
    label="Report history"
  />
  <SideBarItem
    highlight={false}
    icon="help circle"
    label="Help"
  />
  <SideBarItem
    highlight={false}
    icon="comment"
    label="Send feedback"
  />
  <Divider />
  <SideBarFooter />
</Menu>
`;

exports[`renders SideBarHeader with props.title='' 1`] = `
<MenuItem>
  <MenuHeader
    className="side-bar-header"
  />
</MenuItem>
`;

exports[`renders SideBarHeader with sample title 1`] = `
<MenuItem>
  <MenuHeader
    className="side-bar-header"
  >
    SAMPLE TITLE
  </MenuHeader>
</MenuItem>
`;
